### PR TITLE
Add to k8s_raw docs re: vault-encrypted files

### LINKED ISSUE #47259: k8s_raw cannot handle Ansible Vault encrypted files
##### SUMMARY

k8s_raw works fine with raw .yaml files but does not work with Ansible Vault encrypted files, but fails apparently trying to parse them. It does not warn if no vault password is provided like normal modules would.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

k8s_raw

##### ANSIBLE VERSION

```paste below
ansible 2.7.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Jul 13 2018, 13:06:57) [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)]
```

##### CONFIGURATION

```paste below
ANSIBLE_SSH_ARGS(/etc/ansible/ansible.cfg) = -o ForwardAgent=yes
```

##### OS / ENVIRONMENT

CentOS 7 trying to hit Minikube VM

##### STEPS TO REPRODUCE

Generate a .yml file for use with k8s_raw, encrypt with `ansible-vault`, and try to use as `src:`.

playbook:
```yaml
#!/usr/bin/env ansible-playbook

- hosts: localhost
  become: true
  become_user: root
  gather_facts: false
  tasks:
    - include_role:
        name: ../../roles/k8s-has-resource
      vars:
        secret_path: resources/test.yml
        kubeconfig_path: "{{ kubectl_config_host_path }}"
```
role:
```
- name: Resource is present
  k8s:
    state: present
    src: "{{ secret_path }}"
    kubeconfig: "{{ kubeconfig_path }}"
    context: "{{ kubectl_context }}"
    namespace: "{{ namespace }}"
```
resources/test.yml:
```
---
apiVersion: v1
kind: Namespace
metadata:
  name: metrics
```

##### EXPECTED RESULTS

I expect the resource to be successfully verified being present or to be applied.

##### ACTUAL RESULTS

Module fails with `'str' object has no attribute 'get'`.

```paste below
TASK [../../roles/k8s-has-resource : Resource is present] *********************************************************************************************
task path: /w/roles/k8s-has-resource/tasks/main.yml:2
<192.168.99.100> ESTABLISH LOCAL CONNECTION FOR USER: root
<192.168.99.100> EXEC /bin/sh -c 'echo ~root && sleep 0'
<192.168.99.100> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /root/.ansible/tmp/ansible-tmp-1539844152.01-140706276929155 `" && echo ansible-tmp-1539844152.01-140706276929155="` echo /root/.ansible/tmp/ansible-tmp-1539844152.01-140706276929155 `" ) && sleep 0'
Using module file /usr/lib/python2.7/site-packages/ansible/modules/clustering/k8s/k8s.py
<192.168.99.100> PUT /root/.ansible/tmp/ansible-local-5195h_dFx/tmpLoKpuc TO /root/.ansible/tmp/ansible-tmp-1539844152.01-140706276929155/AnsiballZ_k8s.py
<192.168.99.100> EXEC /bin/sh -c 'chmod u+x /root/.ansible/tmp/ansible-tmp-1539844152.01-140706276929155/ /root/.ansible/tmp/ansible-tmp-1539844152.01-140706276929155/AnsiballZ_k8s.py && sleep 0'
<192.168.99.100> EXEC /bin/sh -c '/usr/bin/python2 /root/.ansible/tmp/ansible-tmp-1539844152.01-140706276929155/AnsiballZ_k8s.py && sleep 0'
<192.168.99.100> EXEC /bin/sh -c 'rm -f -r /root/.ansible/tmp/ansible-tmp-1539844152.01-140706276929155/ > /dev/null 2>&1 && sleep 0'
The full traceback is:
Traceback (most recent call last):
  File "/root/.ansible/tmp/ansible-tmp-1539844152.01-140706276929155/AnsiballZ_k8s.py", line 113, in <module>
    _ansiballz_main()
  File "/root/.ansible/tmp/ansible-tmp-1539844152.01-140706276929155/AnsiballZ_k8s.py", line 105, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "/root/.ansible/tmp/ansible-tmp-1539844152.01-140706276929155/AnsiballZ_k8s.py", line 48, in invoke_module
    imp.load_module('__main__', mod, module, MOD_DESC)
  File "/tmp/ansible_k8s_payload_J6uVkn/__main__.py", line 171, in <module>
  File "/tmp/ansible_k8s_payload_J6uVkn/__main__.py", line 167, in main
  File "/tmp/ansible_k8s_payload_J6uVkn/ansible_k8s_payload.zip/ansible/module_utils/k8s/raw.py", line 91, in execute_module
AttributeError: 'str' object has no attribute 'get'

fatal: [localhost]: FAILED! => {
    "changed": false, 
    "module_stderr": "Traceback (most recent call last):\n  File \"/root/.ansible/tmp/ansible-tmp-1539844152.01-140706276929155/AnsiballZ_k8s.py\", line 113, in <module>\n    _ansiballz_main()\n  File \"/root/.ansible/tmp/ansible-tmp-1539844152.01-140706276929155/AnsiballZ_k8s.py\", line 105, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/root/.ansible/tmp/ansible-tmp-1539844152.01-140706276929155/AnsiballZ_k8s.py\", line 48, in invoke_module\n    imp.load_module('__main__', mod, module, MOD_DESC)\n  File \"/tmp/ansible_k8s_payload_J6uVkn/__main__.py\", line 171, in <module>\n  File \"/tmp/ansible_k8s_payload_J6uVkn/__main__.py\", line 167, in main\n  File \"/tmp/ansible_k8s_payload_J6uVkn/ansible_k8s_payload.zip/ansible/module_utils/k8s/raw.py\", line 91, in execute_module\nAttributeError: 'str' object has no attribute 'get'\n", 
    "module_stdout": "", 
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", 
    "rc": 1
}
```

### PR DESCRIPTION
**Pull request by r4j4h**
I didn't read the examples far enough and maybe would not have tried all the examples so I ended up creating https://github.com/ansible/ansible/issues/47259 and learned that definition key with lookup works well and we agreed the docs should say something. :)

+label: docsite_pr

##### SUMMARY

Fixes #47259 

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

k8s_raw

##### ANSIBLE VERSION

```paste below
2.7
```

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by willthames** [comment]
Thanks for doing this - this is good, I think it could be improved further by mentioning in the description of `src` that it doesn't cope with vaulted files

**Comment by r4j4h** [comment]
I would love to, but I can't seem to find where the description of `src` is actually generated.. I only see merge_type in the doc file and I can't seem to find where the rest are being defined. Can you point me in the right direction?

**Comment by r4j4h** [comment]
Of course after I post I think of a more clever way to find it. How does that look? Felt wrong to mash it with the controller file description since only the lookup is needed so I added that as another bullet point instead.

File: lib/ansible/utils/module_docs_fragments/k8s_resource_options.py [hunk 1]
```diff
@@ -38,4 +38,5 @@ class ModuleDocFragment(object):
     - Reads from the local file system. To read from the Ansible controller's file system, use the file lookup
       plugin or template lookup plugin, combined with the from_yaml filter, and pass the result to
       I(resource_definition). See Examples below.
+    - To read Ansible Vault-encrypted files, use the file lookup plugin. See Examples below.
```
Comment (willthames) [inline lib/ansible/utils/module_docs_fragments/k8s_resource_options.py]: I would rephrase this slightly to be: 
```suggestion
    - To read Ansible Vault-encrypted files, use the I(resource_definition) parameter instead along with the C(file) lookup plugin. See Examples below. 
```

File: lib/ansible/modules/clustering/k8s/k8s.py [hunk 2]
```diff
@@ -28,7 +28,7 @@
 description:
   - Use the OpenShift Python client to perform CRUD operations on K8s objects.
   - Pass the object definition from a source file or inline. See examples for reading
-    files and using Jinja templates.
+    files and using Jinja templates and vault-encrypted files.
```
Comment (fabianvf) [inline lib/ansible/modules/clustering/k8s/k8s.py]: Minor nit, I would use `or` rather than `and`.

File: lib/ansible/modules/clustering/k8s/k8s.py [hunk 3]
```diff
@@ -116,7 +116,9 @@
     state: present
     src: /testing/deployment.yml
 
-- name: Read definition file from the Ansible controller file system
+- name: >-
+    Read definition file from the Ansible controller file system
+    or by reading the definition from a file encrypted with Ansible Vault.
```
Comment (fabianvf) [inline lib/ansible/modules/clustering/k8s/k8s.py]: I would make this a separate sentence that notes it will be handled properly if it's vault encrypted, ie

```
Read definition file from the Ansible controller file system. 
If the definition file has been encrypted with Ansible Vault it will automatically be decrypted.
```

**Review by fabianvf — CHANGES_REQUESTED** [review]
Thanks for the PR! I just have a few suggestions for phrasing. I'm not very familiar with vault, so my apologies if my suggestions are technically incorrect.

**Review by fabianvf — APPROVED** [review]
I just realised that I probably shouldn't mark something with `Request changes` while I'm traveling and might not be able to adjust my review in a reasonable amount of time. Switching to `Approve` just so that I don't gum up the wheels of progress.

File: lib/ansible/utils/module_docs_fragments/k8s_resource_options.py [hunk 4]
```diff
@@ -38,4 +38,5 @@ class ModuleDocFragment(object):
...
```
Comment (willthames) [inline lib/ansible/utils/module_docs_fragments/k8s_resource_options.py]: Sorry, my notes on this got lost. I suggest going ahead and mashing the two together. Importantly, you can use the template lookup plugin too.

I'd probably go with something along the lines of:

```
To read from the Ansible controller's file system, including vaulted files, use the file lookup plugin or template lookup plugin,...
```

**Review by willthames — CHANGES_REQUESTED** [review]

**Comment by r4j4h** [comment]
@willthames @fabianvf Great feedback, I will make the suggested changes soon :)

**Comment by willthames** [comment]
Thanks @r4j4h for making the docs better!

**Comment by r4j4h** [comment]
It was truly my pleasure @willthames :) I'm glad I could help and hope to more in time. :sparkling_heart:
